$OpenBSD: patch-js_src_ctypes_libffi_src_x86_freebsd_S,v 1.1.1.1 2013/11/20 20:57:29 landry Exp $
backport devel/libffi/patches/patch-src_x86_freebsd_S
Fix stack alignment on *BSD/i386
http://sourceware.org/ml/libffi-discuss/2013/msg00127.html
https://bugzilla.mozilla.org/show_bug.cgi?id=928381
https://hg.mozilla.org/integration/mozilla-inbound/rev/96420ca47200
Remove in 28
--- js/src/ctypes/libffi/src/x86/freebsd.S.orig	Sat Apr  6 11:20:35 2013
+++ js/src/ctypes/libffi/src/x86/freebsd.S	Sat Apr  6 11:20:51 2013
@@ -49,6 +49,9 @@ ffi_call_SYSV:
 	movl  16(%ebp),%ecx
 	subl  %ecx,%esp
 
+	/* Align the stack pointer to 16-bytes */
+	andl  $0xfffffff0, %esp
+
 	movl  %esp,%eax
 
 	/* Place all of the ffi_prep_args in position  */
